src/session.rs

```diff
@@ -53,13 +53,107 @@ pub struct Session(SessionImp);
 // TODO: UserKnownHostsFile for custom known host fingerprint.
 
 impl Session {
+    /// The method for creating a [`Session`] and externally control the creation of TempDir.
+    ///
+    /// By using the built-in [`SessionBuilder`] in openssh, or a custom SessionBuilder,
+    /// create a TempDir.
+    ///
+    /// # Examples
+    ///
+    /// ```ignore
+    /// # use std::error::Error;
+    /// # #[cfg(feature = "process-mux")]
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), Box<dyn Error>> {
+    ///
+    /// use openssh::{Session, Stdio};
+    /// use openssh_sftp_client::Sftp;
+    ///
+    /// #[derive(Clone, Debug, Default)]
+    /// pub struct CustomSessionBuilder {
+    ///    user: Option<String>,
+    ///    port: Option<String>,
+    ///    password: Option<String>,
+    /// }
+    ///
+    /// let builder = CustomSessionBuilder::default().password("123456");
+    /// let (builder, destination) = builder.resolve("ssh://jon@ssh.thesquareplanet.com:222");
+    /// let tempdir = builder.launch_master(destination).await?;
```

Comment:
I would rather the examples to be much simpler and pass the compilation.

Reply:
That's a good idea.